pkg/demoinfocs/common/equipment.go

```diff
@@ -267,6 +308,25 @@ func initEqElementToName() {
 	eqElementToName[EqDefuseKit] = "Defuse Kit"
 	eqElementToName[EqKnife] = "Knife"
 	eqElementToName[EqUnknown] = "UNKNOWN"
+	eqElementToName[EqKnifeBayonet] = "Bayonet"
+	eqElementToName[EqKnifeCSS] = "CSS Knife"
+	eqElementToName[EqKnifeFlip] = "Flip Knife"
+	eqElementToName[EqKnifeGut] = "Gut Knife"
+	eqElementToName[EqKnifeKarambit] = "Karambit"
+	eqElementToName[EqKnifeM9Bayonet] = "M9 Bayonet"
+	eqElementToName[EqKnifeTactical] = "Tactical Knife"
+	eqElementToName[EqKnifeFalchion] = "Falchion Knife"
+	eqElementToName[EqKnifeSurvivalBowie] = "Survival Bowie Knife"
+	eqElementToName[EqKnifeButterfly] = "Butterfly Knife"
+	eqElementToName[EqKnifePush] = "Push Knife"
+	eqElementToName[EqKnifeCord] = "Cord Knife"
+	eqElementToName[EqKnifeCanis] = "Canis Knife"
+	eqElementToName[EqKnifeUrsus] = "Ursus Knife"
+	eqElementToName[EqKnifeGypsyJackknife] = "Gypsy Jackknife"
+	eqElementToName[EqKnifeOutdoor] = "Outdoor Knife"
+	eqElementToName[EqKnifeStiletto] = "Stiletto Knife"
+	eqElementToName[EqKnifeWidowmaker] = "Widowmaker Knife"
+	eqElementToName[EqKnifeSkeleton] = "Skeleton Knife"
```

Comment:
this would have to be in a new map `knifeTypeToName` - don't forget to implement the `String()` method on `KnifeType`